### ISSUE TITLE
I'd like the env_file to add-to instead of overriding any existing environment variables

### ISSUE THREAD

**Issue by pgjones** [pdm-project/pdm#1235]
The current [implementation](https://github.com/pdm-project/pdm/blob/8be6993c27a92624292e73bb9aa840972bdf5f62/pdm/cli/commands/run.py#L135) overrides any existing environment variables. 

I'd suggest changing to,
```python
if env_file:
    process_env = dotenv.dotenv_values(project.root / env_file, encoding="utf-8")
    process_env.update(os.environ.copy())
else:
    process_env = os.environ.copy()
```
This is how I'd expect it to work, and how dotenv works by default ["By default, load_dotenv doesn't override existing environment variables."](https://saurabh-kumar.com/python-dotenv/).

This is problematic to me as I'd like to set a default in the `env_file` and have the actual environment set values that override it. For example in the `env_file` I have,
```
DATABASE_URL="..."
```
With a script
```
test = {cmd = "pytest", env_file=".env"}
```
However in CI I'd like the github action to set the `DATABASE_URL` and for `pdm run test` to use it rather than the value in the `.env` file.

I'm happy to contribute a PR if this is acceptable.

**Comment by frostming** [comment]
SGTM, go ahead for it.

**Comment by pgjones** [comment]
See #1239

**Comment by pgjones** [comment]
Closed by #1239
